packages/backend/src/payment-method/ilp/service.test.ts

```diff
@@ -0,0 +1,368 @@
+    })
+
+    test('fails if slippage too high', async (): Promise<void> =>
+      withConfigOverride(
+        () => config,
+        { slippage: 101 },
+        async () => {
+          mockRatesApi(exchangeRatesUrl, () => ({}))
+
+          await expect(
+            ilpPaymentService.getQuote({
+              paymentPointer: paymentPointerMap['USD'],
+              receiver: await createReceiver(deps, paymentPointerMap['USD']),
+              debitAmount: {
+                assetCode: 'USD',
+                assetScale: 2,
+                value: 100n
+              }
+            })
+          ).rejects.toBe(Pay.PaymentError.InvalidSlippage)
+        }
+      )())
+
+    test('throws if quote returns invalid maxSourceAmount', async (): Promise<void> => {
+      const ratesScope = mockRatesApi(exchangeRatesUrl, () => ({}))
+
+      const options: StartQuoteOptions = {
+        paymentPointer: paymentPointerMap['USD'],
+        receiver: await createReceiver(deps, paymentPointerMap['USD'])
+      }
+
+      jest.spyOn(Pay, 'startQuote').mockResolvedValueOnce({
+        maxSourceAmount: -1n
+      } as Pay.Quote)
+
+      await expect(ilpPaymentService.getQuote(options)).rejects.toThrow(
+        'Invalid maxSourceAmount in ILP quote'
+      )
+
+      ratesScope.done()
+    })
+
+    test('throws if quote returns invalid minDeliveryAmount', async (): Promise<void> => {
+      const ratesScope = mockRatesApi(exchangeRatesUrl, () => ({}))
+
+      const options: StartQuoteOptions = {
+        paymentPointer: paymentPointerMap['USD'],
+        receiver: await createReceiver(deps, paymentPointerMap['USD'], {
+          incomingAmount: {
+            assetCode: 'USD',
+            assetScale: 2,
+            value: 100n
+          }
+        })
+      }
+
+      jest.spyOn(Pay, 'startQuote').mockResolvedValueOnce({
+        maxSourceAmount: 1n,
+        minDeliveryAmount: -1n
+      } as Pay.Quote)
+
+      await expect(ilpPaymentService.getQuote(options)).rejects.toThrow(
+        'Invalid minDeliveryAmount in ILP quote'
+      )
+
+      ratesScope.done()
+    })
+
+    describe('successfully gets ilp quote', (): void => {
+      describe('with incomingAmount', () => {
+        test.each`
+          incomingAssetCode | incomingAmountValue | debitAssetCode | expectedDebitAmount | exchangeRate | slippage | description
+          ${'EUR'}          | ${100n}             | ${'USD'}       | ${101n}             | ${1.0}       | ${0}     | ${'same currency, no slippage'}
+          ${'USD'}          | ${100n}             | ${'USD'}       | ${102n}             | ${1.0}       | ${0.01}  | ${'same currency, some slippage'}
+          ${'EUR'}          | ${100n}             | ${'USD'}       | ${113n}             | ${0.9}       | ${0.01}  | ${'cross currency, exchange rate < 1'}
+          ${'EUR'}          | ${100n}             | ${'USD'}       | ${51n}              | ${2.0}       | ${0.01}  | ${'cross currency, exchange rate > 1'}
+        `(
```

Comment:
This is a good demonstration of the interledger `Pay` library behaviour. This may look odd, but some of these are programmed into the library, for example, the off-by-one "issue", where an extra 1 unit is added to the debitAmount to account for possible packet loss. 

Some behavior is _weird_ for example, slippage being taken into account for a payment within the same asset. 
We have taken note of this behaviour, and will continue investigating.